docs/DeveloperGuide.md

```diff
@@ -1,34 +1,429 @@
 
-### Value proposition
+The purpose of this developer guide is to aid any curious or interested contributor in developing NUSMaze further by providing more insight on how the features were implemented.
 
-{Describe the value proposition: what problem does it solve?}
+### 1.2. Setting up and getting started 
+Step 1. Ensure that Java 11 and IntelliJ Idea (or your preferred Java IDE) are installed in your computer.  
+Step 2. Fork the NUSMaze repo from [here](https://github.com/AY2021S2-CS2113T-T09-2/tp), and clone the fork into your computer.    
+Step 3. Configure the JDK in IntelliJ Idea to use JDK 11 by following instructions from [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).    
+Step 4. Import the project as a Gradle project.
+Step 5. If you had previously disabled the Gradle plugin, go to `File → Settings → Plugins` to re-enable them.  
+Step 6. Click on Import Project and select the build.gradle file.  
+Step 7. Navigate to the NUSMaze class via the path `src → main → java → seedu.duke → NUSMaze` and right click on it.  
+Step 8. Press run on the `Main()` method of NUSMaze.
 
-## User Stories
+If the set up process had been completed successfully, you should see the following message:  
+![Screenshot 2021-03-25 at 7 03 08 PM](https://user-images.githubusercontent.com/60348727/113017279-e14b9d00-91b1-11eb-8ec3-37c0c3f80475.png)
 
-|Version| As a ... | I want to ... | So that I can ...|
+---------------------------------------------------------------------------------------------
+
+## *2. Design* 
+### 2.1. Architecture
+![img.png](images/architecture.png)
+
+The **Architecture Diagram** above depicts the high-level design of the NUSMaze. You can always refer to this diagram
+to understand how the different components of NUSMaze interact with each other.
+
+The class [`NusMaze`](https://github.com/AY2021S2-CS2113T-T09-2/tp/blob/master/src/main/java/seedu/duke/NusMaze.java) is where the `main()` 
+method belongs and is reponsible for:
+* When the app launches, initialise and connect different components of the NUSMaze in correct sequence.
+* When the app terminates, shut down all the components.
+
+Architecture Components of NUSMaze:
+* [**`UIManager`**](#22-uimanager-component): The user interface of the app
+* [**`Parser`**](#23-parser-component): Processes commands inputted by the user
+* [**`Command`**](#24-command-component): Executes the user command 
+* [**`Router`**](#25-router-component): Searches shortest router
+* [**`Data`**](#26-data-component): Holds the data of the app in memory
+* [**`Storage`**](#27-storage-component): Reads data from and write data to external text files
+
+Explanations on how each component is designed and how it functions are further elaborated in the following 
+chapters of the developer guide.
+
+### 2.2. UIManager Component 
+![img.png](images/ui_design.png)  
```

Comment:
![image](https://user-images.githubusercontent.com/57165946/113230187-14ce1a80-92cb-11eb-80b0-3f41800d336f.png)
Consider putting the method name first and then a ':' to show what the method returns to follow the format from the UML reference diagram provided